src/components/ComponentConfiguration.stories.tsx

```diff
@@ -278,3 +281,63 @@ export const Email: StoryObj<typeof Template> = {
     });
   },
 };
+
+export const NumberField: Story = {
+  render: Template,
+  name: 'type: numberfield',
+
+  args: {
+    component: {
+      id: 'wekruya',
+      type: 'number',
+      key: 'number',
+      label: 'A number field',
+      validate: {
+        required: false,
+      },
+    },
+
+    builderInfo: {
+      title: 'Number',
+      group: 'basic',
+      icon: 'hashtag',
+      schema: {placeholder: ''},
+      weight: 30,
+    },
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    await expect(canvas.getByLabelText('Label')).toHaveValue('A number field');
+    await waitFor(async () => {
+      await expect(canvas.getByLabelText('Property Name')).toHaveValue('aNumberField');
+    });
+    await expect(canvas.getByLabelText('Description')).toHaveValue('');
+    await expect(canvas.getByLabelText('Show in summary')).toBeChecked();
+    await expect(canvas.getByLabelText('Show in email')).not.toBeChecked();
+    await expect(canvas.getByLabelText('Show in PDF')).toBeChecked();
+    await expect(canvas.queryByLabelText('Placeholder')).not.toBeInTheDocument();
+
+    // ensure that changing fields in the edit form properly update the preview
+    const preview = within(canvas.getByTestId('componentPreview'));
+
+    await userEvent.clear(canvas.getByLabelText('Label'));
+    await userEvent.type(canvas.getByLabelText('Label'), 'Updated preview label');
+    expect(await preview.findByText('Updated preview label'));
+
+    const previewInput = preview.getByLabelText<HTMLInputElement>('Updated preview label');
+    await expect(previewInput).toHaveDisplayValue('');
+    await expect(previewInput.type).toEqual('number');
+
+    // Ensure that the manually entered key is kept instead of derived from the label,
+    // even when key/label components are not mounted.
+    const keyInput = canvas.getByLabelText('Property Name');
+    await fireEvent.change(keyInput, {target: {value: 'customKey'}});
```

Comment:
I'll clean up the `await` - it is indeed not required. I remember this being some baby steps with interaction testing :)

If I remember correctly, the `fireEvent` was deliberate because clearing the input and using `userEvent.type` results in a derived key being set from the label because the value of the field is briefly an empty string. This form is essentially simulating a user selecting the current value and typing a replacement value (and I didn't manage to do the latter with testing-library).